### PR TITLE
Polish query processing 

### PR DESCRIPTION
**Pull request by lujiajing1126**
1. Check iterators and series returned from all shards,
2. Fix logs

### CODE REVIEW

File: pkg/query/logical/plan_indexscan_global.go [hunk 1]
```diff
@@ -86,9 +86,10 @@ func (t *globalIndexScan) Execute(ec executor.ExecutionContext) ([]*streamv1.Ele
 	}
 	var elements []*streamv1.Element
 	for _, shard := range shards {
-		elementsInShard, err := t.executeForShard(ec, shard)
-		if err != nil {
-			return elements, err
+		elementsInShard, shardErr := t.executeForShard(ec, shard)
+		if shardErr != nil {
+			// TODO: shall we proceed?
```
Comment (hanahmily) [inline pkg/query/logical/plan_indexscan_global.go:91]: this fail-fast strategy is good to me

File: pkg/query/logical/plan_indexscan_local.go [hunk 2]
```diff
@@ -173,6 +180,10 @@ func (i *localIndexScan) executeInShard(shard tsdb.Shard) ([]tsdb.Iterator, erro
 		return nil, err
 	}
 
+	if seriesList == nil || len(seriesList) == 0 {
```
Comment (hanahmily) [inline pkg/query/logical/plan_indexscan_local.go:183]: `len()` can handle both empty and nil slices,

**Review by hanahmily — COMMENTED** [review]
some nits

File: pkg/query/logical/plan_indexscan_local.go (hunk 2 above)
Comment (lujiajing1126) [reply pkg/query/logical/plan_indexscan_local.go:183]: Thx! But it is strange. Why didn't the linter report this issue?

File: pkg/query/logical/plan_indexscan_global.go (hunk 1 above)
Comment (lujiajing1126) [reply pkg/query/logical/plan_indexscan_global.go]: `TODO` removed

File: pkg/query/logical/plan_indexscan_local.go (hunk 2 above)
Comment (lujiajing1126) [reply pkg/query/logical/plan_indexscan_local.go]: Fixed

**Review by hanahmily — APPROVED** [review]

**Pull request merged**